bootloader/grub.go

```diff
@@ -55,16 +55,16 @@ type grub struct {
 // newGrub create a new Grub bootloader object
 func newGrub(rootdir string, opts *Options) Bootloader {
 	g := &grub{rootdir: rootdir}
-	if opts != nil && (opts.Recovery || opts.NoSlashBoot) {
+	if opts != nil {
+		g.uefiRunKernelExtraction = opts.Role == RoleRunMode
```

Comment:
mmm I think this merits an explanation that we only extract the kernel on run mode, and the recovery kernel is loaded from the squashfs and doesn't need to be extracted, because as it is it's a bit opaque as to why only run mode has uefiRunKernelExtraction set, well I mean it's clear since "Run" is in the name of the parameter, but still I would like to see a comment just to the effect of "only extract kernels in run mode"

Reply:
added a comment